drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -1002,10 +1002,14 @@ static int nrf5_configure(const struct device *dev,
 		break;
 #endif /* CONFIG_IEEE802154_NRF5_MULTIPLE_CCA */
 
-	case IEEE802154_CONFIG_RX_ON_WHEN_IDLE:
+	case IEEE802154_CONFIG_RX_ON_WHEN_IDLE: {
 		nrf_802154_rx_on_when_idle_set(config->rx_on_when_idle);
 		nrf5_data.rx_on_when_idle = config->rx_on_when_idle;
-		break;
+
+		if (config->rx_on_when_idle == false) {
+			ARG_UNUSED(nrf_802154_sleep_if_idle());
```

Comment:
```suggestion
			(void)nrf_802154_sleep_if_idle();
```
I believe it's quite uncommon to use `ARG_UNUSED` in this case as it usually documents unused function arguments.